airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java

```diff
@@ -72,13 +73,43 @@ public void ensure(JsonNode schemaJson, JsonNode objectJson) throws JsonValidati
         objectJson.toPrettyString()));
   }
 
+  private static JsonReferenceProcessor getProcessor() {
+    // JsonReferenceProcessor follows $ref in json objects. Jackson does not natively support
+    // this.
+    final JsonReferenceProcessor jsonReferenceProcessor = new JsonReferenceProcessor();
+    jsonReferenceProcessor.setMaxDepth(-1); // no max.
+
+    return jsonReferenceProcessor;
+  }
+
+  /**
+   * Get JsonNode for an object defined as the main object in a JsonSchema file. Able to create the
+   * JsonNode even if the the JsonSchema refers to objects in other files.
+   *
+   * @param schemaFile - the schema file
+   * @return schema object processed from across all dependency files.
+   */
   public static JsonNode getSchema(final File schemaFile) {
     try {
-      // JsonReferenceProcessor follows $ref in json objects. Jackson does not natively support
-      // this.
-      final JsonReferenceProcessor jsonReferenceProcessor = new JsonReferenceProcessor();
-      jsonReferenceProcessor.setMaxDepth(-1); // no max.
-      return jsonReferenceProcessor.process(schemaFile);
+      return getProcessor().process(schemaFile);
+    } catch (IOException | JsonReferenceException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Get JsonNode for an object defined in the "definitions" section of a JsonSchema file. Able to
+   * create the JsonNode even if the the JsonSchema refers to objects in other files.
+   *
+   * @param schemaFile - the schema file
+   * @param definitionStructName - get the schema from a struct defined in the "definitions" section
+   *        of a JsonSchema file (instead of the main object in that file).
+   * @return schema object processed from across all dependency files.
+   */
+  public static JsonNode getSchema(final File schemaFile, String definitionStructName) {
```

Comment:
Changing the structure of the airbyte_message / airbyte_protocol file means that we need to get the schema from an object that is not the main object in the jsonschema file. So needed to rejigger this internal lib to handle the case where we want to put the jsonschema of an object defined in "definitions" instead of the main file.